File: News_Service/src/main/java/com/dunice/projectNews/exception/CustomException.java
package com.dunice.projectNews.exception;

import com.dunice.projectNews.util.ServerErrorCodes;
import lombok.Getter;

@Getter
public class CustomException extends RuntimeException {

    private final ServerErrorCodes serverErrorCodes;

    private final String message;

    public CustomException(ServerErrorCodes serverErrorCodes) {
        this.serverErrorCodes = serverErrorCodes;
        message = serverErrorCodes.getMessage();
    }
}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/exception/CustomException.kt
package com.dunice.advancedServiceKotlin.exception

import com.dunice.advancedServiceKotlin.util.ServerErrorCodes
import lombok.Getter

class CustomException(private val serverErrorCodes: ServerErrorCodes) : RuntimeException() {
    override val message: String = serverErrorCodes.message
}
